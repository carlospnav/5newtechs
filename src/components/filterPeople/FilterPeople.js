import styled from "styled-components"
import css from "theme/variables"
import { flexGroup } from "theme/utils"


const Input = styled.input.attrs({
  type: "text",
  placeholder: "Find your character"
})`
  font-family: ${css.font.roboto};
  font-size: 18px;
  font-weight: ${css.fontWeight.regular};
  letter-spacing: 0.7px;
  height: 50px;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #818181;
  background-color: black;
  padding-left: 20px;
  margin: 0 20px 40px 20px;
 
  &::placeholder{
    color: white;
  }

  @media (min-width: 500px){
    width: 460px;
  }
`
const SearchIcn = styled.img.attrs({
  src: "/static/search/icn-search.svg",
  alt: "Search Icon"
})`
  position: absolute;
  right: 20px;
  top: 16px;

  @media (min-width: 500px){
    right: 40px;
  }
`

const ErrorMessage = styled.p`
  color: white;
  text-align: center;
  max-width: 300px;
  margin: 0 0 200px 0;
  font-family: ${css.font.error};
`

const Container = styled.div`
  ${flexGroup("column", "center", "center")}
  position: relative;
`

export default function({error, onChange}){
  return(
    <Container>
      <Input onChange={onChange}/>
      <SearchIcn />
      {error && <ErrorMessage>{error}</ErrorMessage> }
    </Container>
  )
}