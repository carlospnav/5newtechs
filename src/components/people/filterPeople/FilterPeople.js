import styled from "styled-components"
import css from "theme/variables"
import { flexGroup } from "theme/utils"
import ErrorMessage from "common/errorMessage/ErrorMessage"
import SearchIcn from "common/searchIcon/SearchIcon"


export const Input = styled.input.attrs({
  type: "text",
  placeholder: "Find your character"
})`
  font-family: ${css.font.roboto};
  font-size: 18px;
  font-weight: ${css.fontWeight.regular};
  letter-spacing: 0.7px;
  height: 50px;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #818181;
  background-color: black;
  padding-left: 20px;
  margin: 0 20px 40px 20px;
  color: white;
 
  &::placeholder{
    color: white;
  }

  @media (min-width: 500px){
    width: 460px;
  }
`

const Container = styled.div`
  ${flexGroup("column", "center", "center")}
  position: relative;
`

export default function({error, onChange}){
  return(
    <Container>
      <Input onChange={onChange}/>
      <SearchIcn />
      {error && <ErrorMessage>{error}</ErrorMessage> }
    </Container>
  )
}