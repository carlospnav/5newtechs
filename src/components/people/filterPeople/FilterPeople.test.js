import { shallow } from "enzyme"
import FilterPeople, { Input } from "./FilterPeople"
import "jest-styled-components"

/* I'm not testing for a search icon because I figured the 
snapshots already expect an image. */
describe("<FilterPeople>", () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<FilterPeople />))
  afterEach(() => wrapper = undefined)

  it("should render a FilterPeople component with no errors properly.", () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it("should render a FilterPeople component with an error message properly.", () => {
    wrapper.setProps({error: "There was an error!"})
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it("should render the Input component defined in its file properly.", () => {
    expect(wrapper.find(Input).dive()).toMatchSnapshot()
  })
})