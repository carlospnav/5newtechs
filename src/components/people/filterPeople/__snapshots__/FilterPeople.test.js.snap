// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPeople> should render a FilterPeople component with an error message properly. 1`] = `
.c0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

<div
  className="c0"
>
  <styled.input />
  <styled.img />
  <styled.span>
    There was an error!
  </styled.span>
</div>
`;

exports[`<FilterPeople> should render a FilterPeople component with no errors properly. 1`] = `
.c0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

<div
  className="c0"
>
  <styled.input />
  <styled.img />
</div>
`;

exports[`<FilterPeople> should render the Input component defined in its file properly. 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 18px;
  font-weight: 300;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  height: 50px;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #818181;
  background-color: black;
  padding-left: 20px;
  margin: 0 20px 40px 20px;
  color: white;
}

.c0::-webkit-input-placeholder {
  color: white;
}

.c0::-moz-placeholder {
  color: white;
}

.c0:-ms-input-placeholder {
  color: white;
}

.c0::placeholder {
  color: white;
}

@media (min-width:500px) {
  .c0 {
    width: 460px;
  }
}

<input
  className="c0"
  placeholder="Find your character"
  type="text"
/>
`;
